src/package/compressed.rs

```diff
@@ -47,76 +47,112 @@ impl Package {
     /// This intentionally uses a [`BTreeMap`] to ensure that the list of files is sorted
     /// lexicographically. This ensures a reproducible output.
     pub fn create(mut manifest: Manifest, files: BTreeMap<PathBuf, Bytes>) -> miette::Result<Self> {
+        // Create a new conforming manifest if the edition is unknown
         if manifest.edition == Edition::Unknown {
-            manifest = Manifest::new(manifest.package, manifest.dependencies);
+            manifest = Manifest::new(manifest.package.clone(), manifest.dependencies.clone());
         }
 
+        // Ensure the manifest has a package declaration
         if manifest.package.is_none() {
             return Err(miette!(
-                "failed to create package, manifest doesnt contain a package declaration"
+                "failed to create package, manifest doesn't contain a package declaration"
             ));
         }
 
         let mut archive = tar::Builder::new(Vec::new());
 
+        // Add original and resolved manifests
+        Self::add_manifest_to_archive(
+            &mut archive,
+            &manifest.for_publishing()?,
```

Comment:
```suggestion
            &manifest.resolve()?,
```

I would (maybe) consider adding another manifest new type which holds resolved manifests only and is only constructable by the above fn; that way you can differentiate by the state of the manifest (ie. containing aliases and resolved)

Reply:
f77d2faf2d04a4765aace473cea58f2ed5a09cd0